### ISSUE TITLE
MSSQL docker container keeps restarting

### ISSUE THREAD

**Issue by jakubgs** [bitwarden/server#553]
I keep seeing my MSSQL container restarting and logs showing:
```log
Captured a dump with paldumper
This program has encountered a fatal error and cannot continue running at Tue Aug 20 20:45:10 2019
The following diagnostic information is available:

       Reason: 0x00000002
       Status: 0xd000000d
   Parameters: 0x200cafe00
               [0] 0x10010
   Stacktrace: 000000006bd313f9 000000006bc5f4ed 000000006bc02f19
               000000006bd6cf48 0000000200dd13c5
      Process: 29 - sqlservr
       Thread: 64 (application thread 0x80)
  Instance Id: c24be5bc-5624-40cf-9bbe-cbb009ca4340
     Crash Id: a7834637-2dda-484e-80f6-12b78277a9dd
  Build stamp: b2ce95e3a6684060d1d3bb1264841a1a8fc1d501ed6d1cfdb89cfcdde8048253
 Distribution: Ubuntu 16.04.6 LTS
   Processors: 2
 Total Memory: 4136538112 bytes
    Timestamp: Tue Aug 20 20:45:10 2019

Ubuntu 16.04.6 LTS
Capturing core dump and information to /var/opt/mssql/log...
dmesg: read kernel buffer failed: Operation not permitted
No journal files were found.
No journal files were found.
Attempting to capture a dump with paldumper
Captured a dump with paldumper
```
I have tried setting permissions on the container folders to `777` but that didn't fix anything, so I don't think it's permissions. There is no clear error in the logs.

**Comment by jakubgs** [comment]
The `core.sqlservr.08_22_2019_03_06_25.30.json` contains this:
```json
{
    "reason": "0x00000002",
    "processName": "sqlservr",
    "pid": "30",
    "instanceId": "f67a75d3-d529-4d95-adcd-e0a4667566b2",
    "crashId": "45291cdb-7445-4736-8eeb-c100c4dff031",
    "threadId": "65",
    "libosThreadId": "0x80",
    "buildStamp": "b2ce95e3a6684060d1d3bb1264841a1a8fc1d501ed6d1cfdb89cfcdde8048253",
    "status": "0xd000000d",
    "parameters": [
        {
            "value": "0x10010"
        }
    ],
    "libosStack": [
        "0x000000006b3313f9",
        "0x000000006b25f4ed",
        "0x000000006b202f19",
        "0x000000006b36cf48",
        "0x0000000380dd13c5"
    ],
    "last_errno": "2",
    "last_errno_text": "No such file or directory",
    "distribution": "Ubuntu 16.04.6 LTS",
    "processors": "2",
    "total_memory": "4136538112",
    "timestamp": "Thu Aug 22 03:06:25 2019"
```
Yes, it's cut off like that. No clue what `No such file or directory` is referring to.

**Comment by jakubgs** [comment]
Found something inside of `thread_information.log`:
```
cat: /proc/30/stack: Permission denied
```
Not sure if that's the actual error or an error that happened after the crash when processing it.

**Comment by jakubgs** [comment]
Gist of `info.log` from the crash: https://gist.github.com/jakubgs/2327c2749392525446c9981acf954fa9

I'm baffled, I have no idea what the actual error is, it just keeps crashing and restarting.

**Comment by jakubgs** [comment]
The weirdest thing is that sometimes when i remove all bitwarden files and re-create all container from scratch it works, and sometimes it keeps restarting forever. Makes no sense.

**Comment by kspearrin** [comment]
maybe try using docker volume option in ./bwdata/config.yml?

**Comment by jakubgs** [comment]
Not sure what you're referring to:
```
 /opt/bitwarden % sudo find | grep bwdata    
 /opt/bitwarden % sudo find | grep config.yml
 /opt/bitwarden % sudo find | grep config.yml | wc -l
0
```

**Comment by kspearrin** [comment]
See install docs here: https://help.bitwarden.com/article/install-on-premise/#post-install-environment-configuration

**Comment by jakubgs** [comment]
Oh, you're referring to setup with the script. I didn't use that. I downloaded the stub archive and configure it using this ansible role I've created: https://github.com/status-im/infra-role-bitwarden
So there is no `config.yml`.

**Comment by kspearrin** [comment]
In that case, try swapping the mssql mapped volume for a docker volume.

**Comment by jakubgs** [comment]
Okay, this makes NO sense to me, but instead of using the ansible `docker_compose` module like this:
```yaml
- name: BitWarden | Compose the containers
  docker_compose:
    project_name: 'bitwarden'
    project_src: '{{ bitwarden_stub_path }}'
    files: 'docker/docker-compose.yml'
    pull: true
    state: '{{ bitwarden_compose_state }}'
    recreate: '{{ bitwarden_compose_recreate }}'
    restarted:  '{{ bitwarden_compose_restart }}'
```
I used the `docker-compose` command directly like this:
```yaml
- name: BitWarden | Compose the containers
  command: docker-compose -f ./docker/docker-compose.yml up -d
  args:
    chdir: '{{ bitwarden_stub_path }}'
```
And even though the result should be the same, the MSSQL container starts this way just fine.
Make no sense, but okay... I'm closing this, maybe some will stumble on the same issue.

**Comment by alegiovannelli** [comment]
I noticed recently that I have the same problem described with the production of many log files but especially in / bwdata / logs / mssql there are several directories (core.sqlservr.09_27_2019_00_xx_xx.xx.d ....) having inside a very large core.sqlservr.31.gdmp file, approximately 1GB.

I saved about 200 passwords in my vault, for a log file it seems to be excessive!

My installation is the one indicated in the documentation on a debian 10 distribution in a Proxmox VM.

Any suggestions?

Thanks
